mesonbuild/coredata.py

```diff
@@ -412,6 +389,8 @@ def get_nondefault_buildtype_args(self) -> T.List[T.Union[T.Tuple[str, str, str]
             return []
         actual_opt = self.optstore.get_value_for('optimization')
         actual_debug = self.optstore.get_value_for('debug')
+        assert isinstance(actual_opt, str) # for mypy
+        assert isinstance(actual_opt, bool) # for mypy
```

Comment:
Something is off here. I guess it should be

```suggestion
        assert isinstance(actual_opt, str) # for mypy
        assert isinstance(actual_debug, bool) # for mypy
```